src/main/java/uk/gov/justice/probation/courtcaseservice/application/WebClientConfig.java

```diff
@@ -24,26 +23,33 @@
     @Value("${offender-assessments-api.base-url}")
     private String assessmentsApiBaseUrl;
 
-    @Bean(name = "communityApiClient")
-    public RestClientHelper getCommunityApiClient(WebClient communityWebClient) {
+    @Value("${nomis-oauth.base-url}")
+    private String oauthApiBaseUrl;
+
+    @Bean
+    public RestClientHelper communityApiClient(WebClient communityWebClient) {
         return new RestClientHelper(communityWebClient, "community-api-client", disableAuthentication);
     }
 
-    @Bean(name = "assessmentsApiClient")
-    public RestClientHelper getAssessmentsApiClient(WebClient assessmentsWebClient) {
+    @Bean
+    public RestClientHelper assessmentsApiClient(WebClient assessmentsWebClient) {
         return new RestClientHelper(assessmentsWebClient, "offender-assessments-api-client", disableAuthentication);
     }
 
     @Bean
     public WebClient communityWebClient(OAuth2AuthorizedClientManager authorizedClientManager) {
         return webClientFactory(communityApiBaseUrl, authorizedClientManager);
     }
-
     @Bean
     public WebClient assessmentsWebClient(OAuth2AuthorizedClientManager authorizedClientManager) {
         return webClientFactory(assessmentsApiBaseUrl, authorizedClientManager);
     }
 
+    @Bean
+    public WebClient oauthWebClient(OAuth2AuthorizedClientManager authorizedClientManager) {
+        return webClientFactory(oauthApiBaseUrl, authorizedClientManager);
+    }
+
     private WebClient webClientFactory(String baseUrl, OAuth2AuthorizedClientManager authorizedClientManager) {
```

Comment:
in CCM, we overwrote the web client with configurable timeouts - see code snippet below. I think we should adopt this now. The delays we have seen in preprod are much too long for a user.

I think that we should have shorter timeouts for our "pinging" clients. Especially for the read timeout.

`        HttpClient httpClient = HttpClient.create()
            .tcpConfiguration(client ->
                client.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, connectTimeoutMs)
                    .doOnConnected(conn -> conn
                        .addHandlerLast(new ReadTimeoutHandler(readTimeoutMs, TimeUnit.MILLISECONDS))
                        .addHandlerLast(new WriteTimeoutHandler(writeTimeoutMs, TimeUnit.MILLISECONDS))));`